src/api.rs

```diff
@@ -74,6 +74,37 @@ pub enum Registers {
 pub const PAGE_SHIFT: u32 = 12;
 pub const PAGE_SIZE: u32 = 4096;
 
+#[repr(C)]
+#[derive(Debug)]
+pub enum Access {
+    NIL,
+    R,
+    W,
+    X,
+    RW,
+    RX,
+    WX,
+    RWX,
+}
+
+impl From<u8> for Access {
```

Comment:
Afais this is KVM specific so it does not belong in this file. Also, if you need to account for error cases you should implement `TryFrom` instead of `From`